docs/refs/api/platform/sleep.md

```diff
@@ -0,0 +1,92 @@
+#### Sleep
+
+There is only one sleep function in Mbed OS 5.6
+
+```c++
+void sleep();
+```
+
+This function invokes sleep manager, which we introduce below.
+
+The idle loop invokes sleep manager by default. You can overwrite this default behaviour by attaching a different idle hook function pointer.
+
+```c++
+void new_idle_loop()
+{
+    // do nothing
+}
+
+void main()
+{
+    rtos_attach_idle_hook(&new_idle_loop);
+}
+```
+
+##### Sleep modes
+
+There are two available sleep modes:
+
+1. Sleep mode
+
+    The system clock to the core stops until a reset or an interrupt occurs. This eliminates dynamic power that the processor, memory systems and buses use. This mode maintains the processor, peripheral and memory state, and the peripherals continue to work and can generate interrupts.
+
+    You can wake up the processor by any internal peripheral interrupt or external pin interrupt.
+
+2. Deep sleep mode 
+
+    This processor is set up ready for deep sleep, and sent to sleep. This mode has the same sleep features as sleep, and it also powers down peripherals and high speed clocks. This mode maintains all state. You can only wake up the processor by lp ticker, RTC, an external interrupt on a pin or a watchdog timer.
+
+##### Sleep manager
+
+The sleep manager provides API to control sleep modes. Deep sleep might introduce some power savings that can affect an application, for instance high speed clock dependent drivers.
+
+The `DeepSleepLock` class provides RAII object for disabling sleep, or explicit lock/unlock methods. To prevent an application from entering deep sleep, invoke `DeepSleepLock()::lock()` method. As soon as an application is ready for the deep sleep, allow it by invoking the `DeepSleepLock::unlock()` method.
+
+These Mbed OS drivers contain locking deep sleep:
+
+- `Ticker`
+- `Timeout`
+- `Timer`
+- `SPI`
+- `I2C`
+- `CAN`
+- `SerialBase`
+
+### API
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.6/api/DeepSleepLock_8h_source.html)
```

Comment:
I think we should also reference this header:
https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_sleep.h